plugins/outputs/nats/nats_test.go

```diff
@@ -263,3 +268,106 @@ func createStream(t *testing.T, server string, cfg *nats.StreamConfig) {
 	_, err = js.AddStream(cfg)
 	require.NoError(t, err)
 }
+
+func TestWriteWithLayoutIntegration(t *testing.T) {
+	if testing.Short() {
+		t.Skip("Skipping integration test in short mode")
+	}
+	natsServicePort := "4222"
+
+	container := testutil.Container{
+		Image:        "nats:latest",
+		ExposedPorts: []string{natsServicePort},
+		Cmd:          []string{"--js"},
+		WaitingFor:   wait.ForListeningPort(nat.Port(natsServicePort)),
+	}
+
+	natsInstance := &NATS{
+		Name: "telegraf",
+		Jetstream: &StreamConfig{
+			Name:     "my-telegraf-stream",
+			Subjects: []string{"my-subject.>"},
+		},
+		serializer: &influx.Serializer{},
+		Log:        testutil.Logger{},
+	}
+
+	streamConfigCompareFunc := func(t *testing.T, si *jetstream.StreamInfo) {
+		require.Equal(t, "my-telegraf-stream", si.Config.Name)
+		require.Equal(t, []string{"my-subject.>"}, si.Config.Subjects)
+	}
+
+	type testConfig struct {
+		name             string
+		subject          string
+		sendMetrics      []telegraf.Metric
+		expectedSubjects []string
+	}
+	testCases := []testConfig{
+		{
+			name:    "subject layout with tags",
+			subject: "my-subject.metrics.{{ .Name }}.{{ .Tag \"tag1\" }}.{{ .Tag \"tag2\" }}",
+			sendMetrics: []telegraf.Metric{metric.New(
+				"test1",
+				map[string]string{"tag1": "foo", "tag2": "bar"},
+				map[string]interface{}{"value": 1.0},
+				time.Date(2009, time.November, 10, 23, 0, 0, 0, time.UTC),
+			)},
+			expectedSubjects: []string{
+				"my-subject.metrics.test1.foo.bar",
+			},
+		},
+		{
+			name:    "subject layout with field name",
+			subject: "my-subject.metrics.{{ .Tag \"tag1\" }}.{{ .Tag \"tag2\" }}.{{ .Name }}.{{ .Field \"value\" }}",
+			sendMetrics: []telegraf.Metric{metric.New(
+				"test1",
+				map[string]string{"tag1": "foo", "tag2": "bar"},
+				map[string]interface{}{"value": 1.0},
+				time.Date(2009, time.November, 10, 23, 0, 0, 0, time.UTC),
+			)},
+			expectedSubjects: []string{
+				"my-subject.metrics.foo.bar.test1.1",
+			},
+		},
+	}
+
+	require.NoError(t, container.Start(), "failed to start container")
+	defer container.Terminate()
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			natsInstance.Subject = tc.subject
+			server := []string{fmt.Sprintf("nats://%s:%s", container.Address, container.Ports[natsServicePort])}
+			natsInstance.Servers = server
+			require.NoError(t, natsInstance.Init())
+			require.NoError(t, natsInstance.Connect())
+
+			stream, err := natsInstance.jetstreamClient.Stream(t.Context(), natsInstance.Jetstream.Name)
+			require.NoError(t, err)
+			si, err := stream.Info(t.Context())
+			require.NoError(t, err)
+
+			streamConfigCompareFunc(t, si)
+			require.NoError(t, natsInstance.Write(tc.sendMetrics))
+			metricCount := len(tc.sendMetrics)
+
+			foundSubjects := make([]string, 0, metricCount)
+			if natsInstance.Jetstream != nil {
```

Comment:
The `Jetstream` should always be non-nil, shouldn't it?

Reply:
Good catch